resources/ecr-scan-check.py

```diff
@@ -0,0 +1,232 @@
+#!/usr/bin/env python3
+
+import argparse
+import botocore
+import boto3
+import json
+import os
+import sys
+import time
+
+severities = [
+    'INFORMATIONAL',
+    'LOW',
+    'MEDIUM',
+    'HIGH',
+    'CRITICAL',
+]
+
+def log_output(level, message):
+    colours = {
+        "Info": "\033[1;34m",
+        "Error": "\033[0;31m",
+        "Success": "\033[0;32m",
+        "end": "\033[0m",
+    }
+
+    string = "{}{}:{} {}".format(colours[level], level, colours["end"], message)
+    print(string)
+
+def check_environment_variables():
+    required_vars = [
+        'AWS_ACCESS_KEY_ID',
+        'AWS_SECRET_ACCESS_KEY',
+        'AWS_REGION',
+    ]
+
+    missing_vars = []
+    for env_var in required_vars:
+        if os.getenv(env_var) is None:
+            missing_vars.append(env_var)
+
+    if missing_vars:
+        log_output("Error", f"Mandatory environment variable(s) undefined: {', '.join(missing_vars)}")
+        sys.exit(1)
+
+
+def parse_arguments():
+    parser = argparse.ArgumentParser(description="Program arguments and options")
+    parser.add_argument("imagerepo", help="Image repository name")
+    parser.add_argument("imagetag", help="Image tag to filter for")
+
+    return parser.parse_args()
+
+def get_severities(level):
+    if level is None:
+        return severities
+    else:
+        return severities[severities.index(level):]
+
+def ecr_open_session():
+    log_output("Info", "Opening AWS session...")
+    try:
+        session = boto3.Session(
+            aws_access_key_id=os.environ.get("AWS_ACCESS_KEY_ID"),
+            aws_secret_access_key=os.environ.get("AWS_SECRET_ACCESS_KEY"),
+            region_name=os.environ.get("AWS_REGION"),
+        )
+        ecr_client = session.client("ecr")
+
+    except session.exceptions.UnauthorizedException as err:
+        logoutput("Error", "Invalid SSO credentials")
```

Comment:
I would change the error text here as the credentials aren't necessarily SSO. I.e. `SSO` -> `AWS`